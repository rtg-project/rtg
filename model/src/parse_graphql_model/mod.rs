pub mod conversion_error;
pub mod convert_document;
pub mod convert_field;
pub mod convert_graphql_string;
pub mod convert_type_definition;

// Tests
#[cfg(test)]
mod tests {

  use super::*;
  use insta::assert_debug_snapshot;

  #[test]
  fn it_works() {
    let model = convert_graphql_string::convert_graphql_string(
      r#"
      type Person {
        id: ID!
        age: Integer @sql(column: "the_age", type: "int8")
      }
      "#,
    )
    .unwrap();
    assert_debug_snapshot!(model);
  }
}
