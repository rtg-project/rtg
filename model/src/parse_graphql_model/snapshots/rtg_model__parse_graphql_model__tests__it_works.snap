---
source: model/src/parse_graphql_model/mod.rs
expression: model

---
V1 {
    entities: Some(
        [
            DatabaseTable {
                name: Some(
                    "Person",
                ),
                sql_schema_name: None,
                sql_table_name: None,
                graphql_entity_type_name: None,
                graphql_filter_type_name: None,
                graphql_get_single_operation_name: None,
                graphql_get_list_operation_name: None,
                graphql_get_connection_operation_name: None,
                graphql_default_order_by: None,
                graphql_default_first: None,
                graphql_default_offset: None,
                fields: Some(
                    [
                        ScalarDatabaseColumn {
                            name: Some(
                                "id",
                            ),
                            nullable: Some(
                                false,
                            ),
                            sql_type: None,
                            sql_column_name: None,
                            graphql_field_name: None,
                            graphql_type_name: Some(
                                "ID",
                            ),
                            graphql_order_by_asc: None,
                            graphql_order_by_desc: None,
                        },
                        ScalarDatabaseColumn {
                            name: Some(
                                "age",
                            ),
                            nullable: Some(
                                true,
                            ),
                            sql_type: Some(
                                Int8,
                            ),
                            sql_column_name: Some(
                                "the_age",
                            ),
                            graphql_field_name: None,
                            graphql_type_name: Some(
                                "Integer",
                            ),
                            graphql_order_by_asc: None,
                            graphql_order_by_desc: None,
                        },
                    ],
                ),
            },
        ],
    ),
}
